translation/models/translation_services.py

```diff
@@ -0,0 +1,167 @@
+from .translation import TranslationStep, TranslatedLesson
+from django.db import models
+from django.conf import settings
+
+import requests
+import json
+
+
+class GoogleTranslator(object):
+    def __init__(self, obj):
+        self.obj = obj
+
+    def say_hello(self):
+        return "hello, I am Sergey Brin!"
+
+
+class AzureTranslator(object):
+    def __init__(self, obj):
+        self.obj = obj
+
+
+class YandexTranslator(object):
+    base_url = "https://translate.yandex.net/api/v1.5/tr.json/translate"
+    service_name = "YANDEX"
+    api_version = 1.5
+    api_key = settings.YANDEX_API_KEY
+    api_controller = models.ForeignKey(
+        'api_controller.ApiController',
+        on_delete=models.CASCADE,
+        related_name='translation_services'
+    )
+
+    def __init__(self, obj):
+        self.obj = obj
+
+    # :param pk: step's stepik_id
+    # :param lang: step's lang
+    # :returns: TranslationStep object or None
+    def get_step_translation(self, pk, lang, **kwargs):
+        # TODO can we optimize request?
+        if lang is None and TranslationStep.objects.filter(stepik_id=pk).exists():
+            return TranslationStep.objects.filter(stepik_id=pk)
+        elif TranslationStep.objects.filter(stepik_id=pk, lang=lang).exists():
+            return TranslationStep.objects.filter(stepik_id=pk, lang=lang)
+        else:
+            return None
+
+    # :param text: step's text in html format
+    # :param lang: step's lang
+    # :returns: TranslationStep object or None
+    def create_step_translation(self, text, **kwargs):
+        final_url = self.base_url
+        params = ["?{0}={1}".format("key", self.api_key), "&{0}={1}".format("text", text)]
+        for name, value in kwargs.items():
+            params.append("&{0}={1}".format(name, value))
+        print("PARAMS:", params)
+        response = requests.get(final_url + "".join(params)).json()
+        return response['text']
+
+    # :param pk: step's stepik_id
+    # :param new_text: new translation of step's text
+    # :param lang: step's lang
+    # :returns: True or False
+    def update_step_translation(self, pk, lang, new_text):
+        if TranslationStep.objects.filter(pk=pk, lang=lang).exists():
+            step = TranslationStep.objects.filter(pk=pk, lang=lang)[0]
+        else:
+            step = self.create_step_translation(new_text, lang=lang)
+        step.text = new_text
+        step.save()
```

Comment:
this should be only when it is already cached? (the 1st branch of `if` statement)